components/script/dom/window.rs

```diff
@@ -319,10 +319,15 @@ impl Window {
         &self.image_cache_thread
     }
 
+    /// Retrieve unsafely the browsing context
     pub fn browsing_context(&self) -> Root<BrowsingContext> {
         self.browsing_context.get().unwrap()
     }
 
+    pub fn safe_browsing_context(&self) -> Option<Root<BrowsingContext>> {
```

Comment:
The only place this method is used outside of `dom::window` is in `Document::browsing_context`, which checks whether `self.has_browsing_context` is true.

Why is `self.has_browsing_context` ever true without a corresponding browsing context in the document's window?